opencontest/contest/models/user.py

```diff
@@ -28,13 +30,23 @@ def getByName(username: str):
             return userNames[username]
         return None
     
+    @staticmethod
+    def getCurrent(request):
+        if request == None:
+            return None
+        if 'id' in request.COOKIES:
+            return User.get(request.COOKIES['id'])
+        return None
+
+
     def save(self):
-        if self.id == None:
-            if self.username in [user.id for user in User.all()]:
-                return  # don't create duplicate user
-            self.id = self.username
+        # If a user with a username of self.username
+        # does not exists, then add it to the global
+        # data structures
+        if self.id not in [user.username for user in User.all()]:
```

Comment:
I am confused by this code. Why are we checking to see if self.id is not one of the user's usernames?

Reply:
That is confusing... That `self.id` should be `self.username`. I'll go and fix that.